common/optimiser/Optimiser.java

```diff
@@ -87,23 +83,32 @@ public boolean isError() {
         return status == Status.ERROR;
     }
 
-    private boolean constraintsChanged() {
-        return iterate(constraints).anyMatch(c -> !c.isUpToDate);
+    public synchronized void reset() {
+        if (hasSolver) {
+            solver.reset();
+            setSolverModel();
+        } else initialiseSolver();
     }
 
     private void initialiseSolver() {
-        solver = MPSolver.createSolver("SAT");
+        solver = MPSolver.createSolver("SCIP");
+        solver.suppressOutput();
```

Comment:
Doesn't actually suppress _all_ of the output.

Reply:
did we try removing this after these changes? I wonder if we still need it since we generally start in the right solution space now...